packages/client/src/v2-events/components/forms/FormFieldGenerator/FormSectionComponent.tsx

```diff
@@ -269,10 +272,29 @@ export function FormSectionComponent({
     resetForm,
     fieldsWithDotSeparator,
     fieldsToShowValidationErrors,
-    setAllFieldsDirty,
+    validateAllFields,
     showValidationErrors
   ])
 
+  // @TODO: Using deepMerge here will cause e2e tests to fail without noticeable difference in the output.
+  // Address is the only deep value.
+  const completeForm = { ...initialValues, ...form }
+
+  const hasAnyValidationErrors = fieldsWithFormikSeparator.some((field) => {
+    const fieldErrors = errors[field.id]?.errors
+    const hasErrors = fieldErrors && fieldErrors.length > 0
+    return isFieldVisible(field, completeForm) && hasErrors
+  })
+
+  useEffect(() => {
+    if (validateAllFields) {
+      showValidationErrors(fieldsWithDotSeparator)
+      onAllFieldsValidated?.(!hasAnyValidationErrors)
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [validateAllFields])
```

Comment:
This def doesn't feel like the neatest solution, lemme know if you come up with a better one :)